File: src/main/scala/com/unbabel/aggregator/event/Event.scala
package com.unbabel.aggregator.event

import java.sql.Timestamp

import com.fasterxml.jackson.annotation.JsonProperty
import com.unbabel.aggregator.util.Time

case class Event(
                  @JsonProperty("timestamp") timestamp: String,
                  @JsonProperty("translation_id") translationId: String,
                  @JsonProperty("source_language") sourceLanguage: String,
                  @JsonProperty("target_language") targetLanguage: String,
                  @JsonProperty("client_name") clientName: String,
                  @JsonProperty("event_name") eventName: String,
                  @JsonProperty("duration") duration: Int,
                  @JsonProperty("nr_words") nrWords: Int
                ) {

  def getTime: Timestamp = Time.byMinute(timestamp)
}

File: src/main/java/com/unbable/unbable_cli/models/Event.java
package com.unbable.unbable_cli.models;

import com.fasterxml.jackson.annotation.JsonFormat;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.datatype.jsr310.deser.LocalDateTimeDeserializer;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
public class Event {

    @JsonDeserialize(using = LocalDateTimeDeserializer.class)
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "yyyy-MM-dd HH:mm:ss.SSSSSS")
    private LocalDateTime timestamp;

    @JsonProperty("translation_id")
    private String translationId;

    @JsonProperty("source_language")
    private String sourceLanguage;

    @JsonProperty("target_language")
    private String targetLanguage;

    @JsonProperty("client_name")
    private String clientName;

    @JsonProperty("event_name")
    private String eventName;

    @JsonProperty("nr_words")
    private Integer nrWords;

    private Integer duration;
}
